# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/script/ModuleLoader.cpp
# Commit: d2647082c823
# Full Hash: d2647082c823595eddd3e051bd99c3cbfc53ae73
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-24 21:46:01
# Description:
#   Bug 1756550 - propagate webext global to all dynamically imported modules; r=jonco
#   
#   The crashes were occurring due to the missing webext global on dynamically imported modules from a referencing script. There is likely a more elegant way to reintroduce this, but for now this is a quick fix.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139511
# ==============================================================================

diff -r e076ff6e02d1 -r d2647082c823 dom/script/ModuleLoader.cpp
--- a/dom/script/ModuleLoader.cpp	Thu Feb 24 12:18:19 2022 +0000
+++ b/dom/script/ModuleLoader.cpp	Thu Feb 24 12:25:25 2022 +0000
@@ -259,7 +259,6 @@
   nsIURI* baseURL = nullptr;
   nsCOMPtr<Element> element;
   RefPtr<ScriptLoadContext> context;
-  // This will be null in all cases except WebExtensions.
   if (script) {
     options = script->GetFetchOptions();
     baseURL = script->BaseURL();
@@ -286,10 +285,10 @@
           xpc::IsWebExtensionContentScriptSandbox(global->GetGlobalJSObject()));
     }
 
-    options = new ScriptFetchOptions(mozilla::CORS_NONE,
-                                     document->GetReferrerPolicy(), principal);
+    options = new ScriptFetchOptions(
+        mozilla::CORS_NONE, document->GetReferrerPolicy(), principal, global);
     baseURL = document->GetDocBaseURI();
-    context = new ScriptLoadContext(nullptr, global);
+    context = new ScriptLoadContext(nullptr);
   }
 
   RefPtr<ModuleLoadRequest> request = ModuleLoader::CreateDynamicImport(
@@ -488,8 +487,7 @@
 already_AddRefed<ModuleLoadRequest> ModuleLoader::CreateStaticImport(
     nsIURI* aURI, ModuleLoadRequest* aParent) {
   RefPtr<ScriptLoadContext> newContext =
-      new ScriptLoadContext(aParent->GetLoadContext()->mElement,
-                            aParent->GetLoadContext()->mWebExtGlobal);
+      new ScriptLoadContext(aParent->GetLoadContext()->mElement);
   newContext->mIsInline = false;
   // Propagated Parent values. TODO: allow child modules to use root module's
   // script mode.